common/thrift_router.h

```diff
@@ -221,6 +240,16 @@ class ThriftRouter {
     local_client_map_.updateClusterLayout(std::move(layout));
   }
 
+  std::shared_ptr<const ClusterLayout> getClusterLayout() {
```

Comment:
We can simply implement this function like below. It's guaranteed that the lock will be released AFTER reading cluster_layout_.

`
{
folly::RWSpinLock::ReadHolder read_guard(layout_rwlock_);
return cluster_layout_;
}
`